src/cmds/pbsnodes.c

```diff
@@ -244,110 +237,63 @@ encode_to_json(struct batch_status *bstat) {
 					ivalue = (long int)value;
 					if (strcmp(pc, "") == 0) {
 						if (value == ivalue)
-							node = add_json_node(JSON_VALUE, JSON_INT, next->resource, &ivalue);
+							if (add_json_node(JSON_VALUE, JSON_INT, next->resource, &ivalue) == NULL)
+								return 1;
 						else
-							node = add_json_node(JSON_VALUE, JSON_FLOAT, next->resource, &value);
-						if (node == NULL)
-							return 1;
+							if (add_json_node(JSON_VALUE, JSON_FLOAT, next->resource, &value) == NULL)
+								return 1;
 					} else if (str[0] != '0') {
-						node = add_json_node(JSON_VALUE, JSON_STRING, next->resource, str);
-						if (node == NULL)
+						if (add_json_node(JSON_VALUE, JSON_STRING, next->resource, str) == NULL)
 							return 1;
 					}
 				} else {
 					if (str[0] != '0') {
-						node = add_json_node(JSON_VALUE, JSON_STRING, next->resource, str);
-						if (node == NULL)
+						if (add_json_node(JSON_VALUE, JSON_STRING, next->resource, str) == NULL)
```

Comment:
I think this `else` and the corresponding `if` is redundant as we are checking the same by comparing `pc`.

Reply:
The `strcmp(pc,"")` is in the if block, this is the else block. This `else` is required for values that are strings that don't have a number in the beginning. 

resources_assigned only wants the non-zero values printed. so we need to loop through every resource, check if we can convert the value to a double (`value = strtod(str, &pc)`).

If `value != 0`, then we can convert part of the value to a double. We then have to check `pc`, which is a char * pointing to the next character in the resource value. If `strcmp(pc, "") == 0` is true, then `pc` is at the end of the string, and we can safely assume it is a number. Then we add it as a int or a float.

If `strcmp(pc,"") == 0` is false, that means there are more characters past the number we've parsed as a double. For example, `1024kb` will reach this case. Because it wasn't fully parsed as a number, we need to add it as a string. 

*Looking at it now, I think everything in the `if (value) {}` block can be replaced by a singular `add_json_node` call.*

If `value == 0`, we finally enter the else case on 249. This means that `strtod` couldn't parse the string as a number. Therefore, we know it's a string and has to be added, unless it's an empty string; then don't bother.

Reply:
Looks much better now :) Thanks!